src/Poker/Evaluate.hs

```diff
@@ -0,0 +1,277 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StrictData #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -Wall #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-type-defaults #-}
+
+-- | Evaluation of a standard holdem poker hand. 'handRank' works for 5 and 7 card lists.
+--
+module Poker.Evaluate
+  ( -- * ['Card'] Hand Ranking
+    HandRank (..),
+    handRank,
+    allHandRanks,
+    straight,
+    flush,
+    kind,
+    rankCount,
+    suitRanks,
+  )
+where
+
+import Control.Applicative
+import Data.Bool
+import Data.Foldable
+import Data.List (sortOn)
+import qualified Data.Map.Strict as Map
+import Data.Maybe
+import Data.Ord
+import qualified Data.Set as Set
+import Data.Tuple
+import GHC.Exts hiding (toList)
+import GHC.Generics hiding (from, to)
+import Poker.Cards
+import Prelude
+
+-- $setup
+--
+-- >>> import Poker.Cards
+-- >>> import Prettyprinter
+-- >>> let cs = [Card Ace Heart,Card Seven Spade,Card Ten Heart,Card Five Spade,Card Six Club, Card Seven Heart,Card Six Spade]
+-- >>> let cs' = [Card Ten Club, Card Five Spade,Card Ten Heart,Card Five Spade,Card Six Club, Card Seven Heart,Card Six Spade]
+-- >>> let css = [cs, cs']
+--
+-- >>> pretty css
+-- [[Ah, 7s, Th, 5s, 6c, 7h, 6s], [Tc, 5s, Th, 5s, 6c, 7h, 6s]]
+
+-- | 5 card standard poker rankings
+--
+-- >>> handRank <$> css
+-- [TwoPair Seven Six Ace,TwoPair Ten Six Five]
+data HandRank
+  = HighCard Rank Rank Rank Rank Rank
+  | OnePair Rank Rank Rank Rank
+  | TwoPair Rank Rank Rank
+  | ThreeOfAKind Rank Rank Rank
+  | Straight Rank
+  | Flush Rank Rank Rank Rank Rank
+  | FullHouse Rank Rank
+  | FourOfAKind Rank Rank
+  | StraightFlush Rank
+  deriving (Eq, Ord, Show, Generic)
+
+-- | enumeration of all possible HandRanks, in ascending order.
+--
+-- >>> length allHandRanks
+-- 7462
+--
+allHandRanks :: [HandRank]
+allHandRanks =
+  [ HighCard a b c d e
+    | a <- ranks,
+      b <- ranksLT a,
```

Comment:
I wonder whether we should expose `ranksLT` in `Cards`? There's a few usages of that logic in `Cards` currently, and this seems like a useful function...